test/int/integration_suite_test.go

```diff
@@ -71,27 +72,23 @@ const (
 var (
 	// This variable is "global" - as is visible only on the first ginkgo node
 	testEnv *envtest.Environment
+	cfg     *rest.Config
 
 	// These variables are initialized once per each node
+	ctx         context.Context
 	k8sClient   client.Client
 	atlasClient *admin.APIClient
 
 	// These variables are per each test and are changed by each BeforeRun
-	namespace         corev1.Namespace
-	cfg               *rest.Config
+	testNamespace     corev1.Namespace
```

Comment:
Would this not be a concurrency issue. I do not think the namespace can be shared across concurrent tests.